sql/parser/builtins.go

```diff
@@ -943,6 +961,125 @@ func datumToString(datum Datum) (string, error) {
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
 
+var replaceSubRe = regexp.MustCompile(`\\[&1-9]`)
+
+func regexpReplace(s, pattern, to, sqlFlags string) (Datum, error) {
+	pattern, global, err := regexpEvalFlags(pattern, sqlFlags)
+	if err != nil {
+		return nil, err
+	}
+
+	patternRe, err := regexp.Compile(pattern)
```

Comment:
The vagueness of my question likely gives away that I'm not too sure what I'm even talking about, but since `regexReplace` is used through the string builtin functions, I would guess that they treat all parameters equally, so they would have to be evaluated for each row anew.